File: src/main/java/org/radarcns/management/service/catalog/SampleRateConfig.java
package org.radarcns.management.service.catalog;

import com.fasterxml.jackson.annotation.JsonProperty;

public class SampleRateConfig {
    @JsonProperty
    private Double interval;

    @JsonProperty
    private Double frequency;

    @JsonProperty
    private boolean dynamic;

    @JsonProperty
    private boolean configurable;

    public Double getInterval() {
        return interval;
    }

    public Double getFrequency() {
        return frequency;
    }

    public boolean isDynamic() {
        return dynamic;
    }

    public boolean isConfigurable() {
        return configurable;
    }

    @Override
    public String toString() {
        return "SampleRateConfig{interval=" + interval
                + ", frequency=" + frequency
                + ", dynamic=" + dynamic
                + ", configurable=" + configurable
                + '}';
    }
}

File: src/main/java/org/radarbase/management/service/catalog/SampleRateConfig.kt
package org.radarbase.management.service.catalog

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class SampleRateConfig {
    @JsonProperty
    val interval: Double? = null

    @JsonProperty
    val frequency: Double? = null

    @JsonProperty
    val isDynamic = false

    @JsonProperty
    val isConfigurable = false
    override fun toString(): String {
        return ("SampleRateConfig{interval=" + interval
                + ", frequency=" + frequency
                + ", dynamic=" + isDynamic
                + ", configurable=" + isConfigurable
                + '}')
    }
}
